subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/overlap/impl/DefaultOverlappingOutputDetector.java

```diff
@@ -107,6 +108,11 @@ public Boolean visitMissing(MissingFileSnapshot missingSnapshot) {
                     // Otherwise check for newly added broken symlinks and unreadable files
                     return hasNewContent(missingSnapshot, null);
                 }
+
+                @Override
+                public Boolean visitUnreadable(UnreadableSnapshot unreadableSnapshot) {
+                    return hasNewContent(unreadableSnapshot, null);
```

Comment:
Won't `hasNewContent` throw when trying to compare the types? Is this intended for overlapping output detection? It seems like this shouldn't throw when the type did not change, so there is probably a test case missing.